test/project_types/extension/commands/serve_test.rb

```diff
@@ -71,6 +71,114 @@ def test_uses_js_system_to_run_npm_or_yarn_serve_commands_with_shop_argument_for
         run_serve
       end
 
+      def test_raises_exception_if_shop_is_missing
+        ShopifyCli::Shopifolk.stubs(:check).returns(true)
+        ShopifyCli::Tasks::EnsureEnv.stubs(:call)
+        ShopifyCli::Tasks::EnsureDevStore.stubs(:call)
+        ShopifyCli::Feature.stubs(:enabled?).with(:argo_admin_beta).returns(true)
+
+        Extension.specifications.stubs(:valid?).returns(true)
+        Extension.specifications.stubs(:[]).returns(
```

Comment:
You should be able to use `DummySpecifications.build` now that my latest PR has shipped:

```ruby
Extension.specifications.stubs(:[]).returns(DummySpecifications.build)
```